### ISSUE TITLE
A question about the track_weights in track_2d_loss

### ISSUE THREAD

**Issue by ChenYutongTHU** [vye16/shape-of-motion#65]
Dear authors,

Thanks for contributing this repository!

I noticed that when computing the track_2d_loss, a track_weights is intended to modulate the 2d error.

https://github.com/vye16/shape-of-motion/blob/697d462fdaccefda5a91373e9b6047b9dafb9234/flow3d/trainer.py#L376-L382

The track_weights is a multiplication of the tracking confidences and a decayed temporal proximity term.

https://github.com/vye16/shape-of-motion/blob/697d462fdaccefda5a91373e9b6047b9dafb9234/flow3d/trainer.py#L360-L362

Then the predicted 2D tracks and pseudo-GT 2D tracks are forwarded to the loss computer:
https://github.com/vye16/shape-of-motion/blob/697d462fdaccefda5a91373e9b6047b9dafb9234/flow3d/loss_utils.py#L26-L42

However, now the _pred_ and _gt_ have shapes of [N, 2] where N is the number of tracks while _mask_ is shaped [N,Batch-sizextarget_steps]. In default configuration, Batch-size=4, and target_steps=8, so the _mask_ can be [N,32].

Then the loss computation might not really make sense as every track of N is multiplied with all the 32 weights though it should only use 1 out of the 32 weights. Since there is a denominator term summing the mask, the 32 weights are canceled out after all.

https://github.com/vye16/shape-of-motion/blob/697d462fdaccefda5a91373e9b6047b9dafb9234/flow3d/loss_utils.py#L38-L39

I find that setting masks in the following two ways obtains the same value of track_2d_loss.
```
        track_2d_loss = masked_l1_loss(
            pred_tracks_2d[masks_flatten][visibles],
            tracks_2d[visibles],
            mask=track_weights[visibles],
            # or mask=confidences[...,None][visibles]
            quantile=0.98,
        ) / max(H, W)
```

I think that revising this will not impact the method's performance but I'd like to examine if my understanding is correct :)

Thanks for your help!
